magento/magento-cloud-patches
# MCLOUD-5953: Backport Redis patches

### Description


### Fixed Issues (if relevant)
https://jira.corp.magento.com/browse/MCLOUD-5953

### Manual testing scenarios

Need to test for Magento 2.3.3 and 2.3.4
1. Edit `composer.json`
```
"repositories": {
        "repo2": {
            "type": "composer",
            "url": "https://repo.magento.com/"
        },
        "ece-tools": {
            "type": "vcs",
            "url": "https://github.com/magento/ece-tools.git"
        },
        "mcp": {
            "type": "vcs",
            "url": "git@github.com:magento/magento-cloud-patches.git"
        },
        "docker":{
            "type": "vcs",
            "url": "git@github.com:magento/magento-cloud-docker.git"
        }
  },
```
```
"require": {
        "magento/magento-cloud-metapackage": ">=2.3.3 <2.3.4",
        "magento/ece-tools": "dev-develop as 2002.1.99",
        "magento/magento-cloud-patches": "dev-MCLOUD-5953 as 1.0.99"
},
```

2. Run `composer update`
3. Commit changes of `composer.json` and `composer.lock`
4. Push to trigger redeploy
5. Check that frontend works
6. Check `app/etc/env.php`
```
'cache' =>
  array (
    'frontend' =>
    array (
      'default' =>
      array (
        'backend' => 'Cm_Cache_Backend_Redis',
        'backend_options' =>
        array (
          'server' => 'redis.internal',
          'port' => 6379,
          'database' => 1,
        ),
        'id_prefix' => 'f53_',
      ),
      'page_cache' =>
      array (
        'backend' => 'Cm_Cache_Backend_Redis',
        'backend_options' =>
        array (
          'server' => 'redis.internal',
          'port' => 6379,
          'database' => 2,
        ),
        'id_prefix' => 'f53_',
      ),
    ),
  ),
```
7. Add `.magento.env.yaml`
```
stage:
  deploy:
    CACHE_CONFIGURATION:
      _merge: false
      frontend:
        default:
          backend: '\Magento\Framework\Cache\Backend\RemoteSynchronizedCache'
          backend_options:
            remote_backend: '\Magento\Framework\Cache\Backend\Redis'
            remote_backend_options:
              persistent: 0
              server: 'redis.internal'
              database: '1'
              port: '6379'
              password: ''
              compress_data: '1'
            local_backend: 'Cm_Cache_Backend_File'
            local_backend_options:
              cache_dir: '/dev/shm/'
          frontend_options:
            write_control: false
      type:
        default:
          frontend: 'default'
```
8. Commit  `.magento.env.yaml`
9. Push to trigger redeploy
10. Check that frontend works
11. Check `app/etc/env.php`
```
'cache' =>
  array (
    'frontend' =>
    array (
      'default' =>
      array (
        'backend' => '\\Magento\\Framework\\Cache\\Backend\\RemoteSynchronizedCache',
        'backend_options' =>
        array (
          'remote_backend' => '\\Magento\\Framework\\Cache\\Backend\\Redis',
          'remote_backend_options' =>
          array (
            'persistent' => 0,
            'server' => 'redis.internal',
            'database' => '1',
            'port' => '6379',
            'password' => '',
            'compress_data' => '1',
          ),
          'local_backend' => 'Cm_Cache_Backend_File',
          'local_backend_options' =>
          array (
            'cache_dir' => '/dev/shm/',
          ),
        ),
        'frontend_options' =>
        array (
          'write_control' => false,
        ),
      ),
    ),
    'type' =>
    array (
      'default' =>
      array (
        'frontend' => 'default',
      ),
    ),
  ),
```

### Contribution checklist
